examples/nats-sub/main.go

```diff
@@ -52,11 +53,24 @@ func main() {
 	opts := []nats.Option{nats.Name("NATS Sample Subscriber")}
 	opts = setupConnOptions(opts)
 
+	if *userCreds != "" && *nkeyFile != "" {
+		log.Fatal("specify -seed or -creds")
+	}
+
 	// Use UserCredentials
 	if *userCreds != "" {
 		opts = append(opts, nats.UserCredentials(*userCreds))
 	}
 
+	// Use Nkey authentication.
+	if *nkeyFile != "" {
+		opt, err := nats.NkeyOptionFromSeed(*nkeyFile)
```

Comment:
Wonder if we can let err processing just happen in Connect to keep examples a bit shorter?

Reply:
tbh,  I think that examples should do things exactly as we want folks to do them. If what you are saying is that setting an option shouldn't return an error until connect then we should change that api instead.